util.py

```diff
@@ -2,140 +2,162 @@
+                    attr.strip('_'), validType.__name__))
+    else:
+        def setMethod(self, val):
+            if colltype and all([type(i) == validType for i in val]):
+                setattr(self, attr, val)
+            else:
+                raise AutomataError('%s must contain only items of type: %s.' % (
+                    attr.strip('_'), validType.__name__))
+
+    return setMethod
+
 
 def genAddMethod(attr, validType):
-  if type(validType) is type:
-    def addMethod(self, item):
-      if type(item) is validType:
-        getattr(self, attr).add(item)
-      else:
-        raise AutomataError, '%s must contain only items of type: %s, not %s.' % (attr.strip('_'), validType.__name__, type(item).__name__)
-  else:
-    def addMethod(self, item):
-      if validType(item):
-        getattr(self, attr).add(item)
-      else:
-        raise AutomataError, 'Attempted to set an invalid  %s.' % attr.strip('_')
-  return addMethod
+    if type(validType) is type:
+        def addMethod(self, item):
+            if type(item) is validType:
+                getattr(self, attr).add(item)
+            else:
+                raise AutomataError('%s must contain only items of type: %s, not %s.' % (
+                    attr.strip('_'), validType.__name__, type(item).__name__))
+    else:
+        def addMethod(self, item):
+            if validType(item):
+                getattr(self, attr).add(item)
+            else:
+                raise AutomataError(
+                    'Attempted to set an invalid  %s.' % attr.strip('_'))
+    return addMethod
+
 
 def genRemMethod(attr, validType):
-  def remItemMethod(self, which):
+    def remItemMethod(self, which):
+        try:
+            if type(which) is validType:
+                getattr(self, attr).remove(which)
+            else:
+                raise AutomataError('%s contains only items of type %s. Attempted to remove a %s' % (
+                    attr.strip('_'), validType.__name__, type(val).__name__))
+        except:
+            raise AutomataError(
+                'Attempted to remove a non-existant item from %s.' % attr.strip('_'))
+    return remItemMethod
+
+# High-level automata usage
+
+
+def renderMachine(fa, outfile='temp.png', format='png', size=None):
+    gvcode = fa.toGraphViz(size)
+
+    f = popen(r'dot -T%s -o"%s"' % (format, outfile), 'w')
     try:
-      if type(which) is validType:
-        getattr(self, attr).remove(which)
-      else:
-        raise AutomataError, '%s contains only items of type %s. Attempted to remove a %s' % (attr.strip('_'), validType.__name__, type(val).__name__)
+        f.write(gvcode)
     except:
-      raise AutomataError, 'Attempted to remove a non-existant item from %s.' % attr.strip('_')
-  return remItemMethod
-
-#High-level automata usage
-def renderMachine(fa, outfile = 'temp.png', format = 'png', size=None):
-  gvcode = fa.toGraphViz(size)
+        raise Exception
```

Comment:
This is a functional difference - to match previous behavior, it should catch and re-raise rather than creating a new exception.

Reply:
I think it is correct now.